sources/dicom/large_image_source_dicom/assetstore/dicomweb_assetstore_adapter.py

```diff
@@ -104,20 +107,161 @@ def deleteFile(self, file):
+            # However, we can't call that since we don't have a great way of
+            # determining the DICOM file size without downloading the whole thing.
+            # FIXME: call that function if we find a way to determine file size.
+
+        # Create the URL
+        url = client._get_instances_url(
+            _Transaction.RETRIEVE,
+            study_uid,
+            series_uid,
+            instance_uid,
         )
 
+        # Build the headers
+        transfer_syntax = '*'
+        accept_parts = [
+            'multipart/related',
+            'type="application/dicom"',
+            f'transfer-syntax={transfer_syntax}',
+        ]
+        headers = {
+            'Accept': '; '.join(accept_parts),
+        }
+
+        def stream():
+            # Perform the request
+            response = client._http_get(url, headers=headers, stream=True)
+            for chunk in self._stream_retrieve_instance_response(response):
+                yield chunk
+
+        return stream
+
+    def _extract_media_type_and_boundary(self, response):
+        content_type = response.headers['content-type']
+        media_type, *ct_info = [ct.strip() for ct in content_type.split(';')]
+        boundary = None
+        for item in ct_info:
+            attr, _, value = item.partition('=')
+            if attr.lower() == 'boundary':
+                boundary = value.strip('"').encode()
+                break
+
+        return media_type, boundary
+
+    def _stream_retrieve_instance_response(self, response):
+        # The first part of this function was largely copied from dicomweb-client's
+        # _decode_multipart_message() function. But we can't use that function here
+        # because it relies on reading the whole DICOM file into memory. We want to
+        # avoid that and stream in chunks.
+
+        # Split the content-type to find the media type and boundary.
+        media_type, boundary = self._extract_media_type_and_boundary(response)
+        if media_type.lower() != 'multipart/related':
+            msg = f'Unexpected media type: "{media_type}". Expected "multipart/related".'
+            raise ValueError(msg)
+
+        # Ensure we have the multipart/related boundary.
+        # The beginning boundary and end boundary look slightly different (in my
+        # examples, beginning looks like '--{boundary}\r\n', and ending looks like
+        # '\r\n--{boundary}--'). But we skip over the beginning boundary anyways
+        # since it is before the message body. An end boundary might look like this:
+        # \r\n--50d7ccd118978542c422543a7156abfce929e7615bc024e533c85801cd77--
+        if boundary is None:
+            content_type = response.headers['content-type']
+            msg = f'Failed to locate boundary in content-type: {content_type}'
+            raise ValueError(msg)
+
+        # Both dicomweb-client and requests-toolbelt check for
+        # the ending boundary exactly like so:
+        ending = b'\r\n--' + boundary
```

Comment:
For reference, [here](https://github.com/requests/toolbelt/blob/c62d9774a3b1469f609bdc5a213f86c714e059bf/requests_toolbelt/multipart/decoder.py#L137-L149) is where `requests-toolbelt` does this. Note the lines:

```python
boundary = b''.join((b'--', self.boundary))
```

and 

```python
parts = content.split(b''.join((b'\r\n', boundary)))
```

So it is essentially `b'\r\n--' + self.boundary`.

And [here](https://github.com/ImagingDataCommons/dicomweb-client/blob/39dc8a799fac508756d9df57cf16fea51d54a9c3/src/dicomweb_client/web.py#L674-L675) is the same kind of thing in `dicomweb_client`.